sdk/core/azure-core/tests/test_request_id_policy.py

```diff
@@ -26,3 +26,29 @@ def test_request_id_policy_fix_on_demand_id():
     pipeline_request.context.options['request_id'] = test_request_id
     request_id_policy.on_request(pipeline_request)
     assert request.headers["x-ms-client-request-id"] == test_request_id
+
+def test_request_id_policy_auto_id():
+    """Test policy with no other policy and happy path"""
+    request_id_policy = RequestIdPolicy()
+    request = HttpRequest('GET', 'http://127.0.0.1/')
+    pipeline_request = PipelineRequest(request, PipelineContext(None))
+    request_id_policy.on_request(pipeline_request)
+    assert request.headers["x-ms-client-request-id"]
+
+def test_request_id_policy_no_id():
+    """Test policy with no other policy and happy path"""
+    request_id_policy = RequestIdPolicy(auto_request_id=False)
+    request = HttpRequest('GET', 'http://127.0.0.1/')
+    pipeline_request = PipelineRequest(request, PipelineContext(None))
+    request_id_policy.on_request(pipeline_request)
+    assert not request.headers["x-ms-client-request-id"]
+
+def test_request_id_policy_set_request_id():
+    """Test policy with no other policy and happy path"""
+    test_request_id = 'test_request_id'
+    request_id_policy = RequestIdPolicy()
+    request_id_policy.set_request_id(test_request_id)
+    request = HttpRequest('GET', 'http://127.0.0.1/')
+    pipeline_request = PipelineRequest(request, PipelineContext(None))
+    request_id_policy.on_request(pipeline_request)
+    assert request.headers["x-ms-client-request-id"] == test_request_id
```

Comment:
To be comprehensive this should test that an explicitly set request id takes precedence in *both* auto-id cases, i.e. on *and* off. Would suggest  `pytest.mark.parametrize` for this. 

Then, there should be a test for a request id in `request.context.options` that should validate that a request id set that way overrides *all* the previous combinations. `pytest.mark.parametrize` can also be helpful here.